backend/backend/views.py

```diff
@@ -34,15 +34,7 @@ def post(self, request, *args, **kwargs):
         serializer = self.serializer_class(data=request.data, context={"request": request})
         serializer.is_valid(raise_exception=True)
         user = serializer.validated_data["user"]
-        if settings.TOKEN_STRATEGY == "reuse":  # nosec
-            token, created = BearerToken.objects.get_or_create(user=user)
-            # handle_token_expiration will check whether the token is expired
-            # and will generate a new one if necessary
-            is_expired, token = bearer_token_utils.handle_token_expiration(token)
-        else:
-            # token should be new each time, remove the old one
-            BearerToken.objects.filter(user=user).delete()
-            token = BearerToken.objects.create(user=user)
+        token = BearerToken.objects.create(user=user, expiry=request.GET.get("expiry"), note=request.GET.get("note"))
```

Comment:
Why do we use request.GET in this POST request?

Reply:
I can't explain why but if I use POST my parameters passed in the URL are then ignored

Reply:
Right, Django calls them `request.GET` and `request.POST` but they maybe should be called `request.url_parameters` and `request.body_parameters`.

Basically we copied the code from `AuthenticatedBearerToken.get` and this is why this classes mixes URL and body parameters. It doesn't necessarily _not_ make sense because the URL parameters are token settings while the body params are authentication -- plus that way the token params are the same for both `/api-token-auth` and `/api-token`.